taojieTan/assignments/twopointers/lc1/lc1.py

```diff
@@ -4,13 +4,42 @@
 # Write a function to return the indices of the two numbers (i.e. the pair) such that they add up to the given target.
 
 def pair_with_targetsum(arr, target_sum):
+  #input: an sorted integer array (positive and/or negative), a target sum integer 
+  #output: a pair of indices 
+  #edge cases: 1. empty array -> None
+  #            2. no such pair adds up to the target sum -> None
+
+  if len(arr) == 0:
+    return None
+
+  #initialize two pointers to keep track of indices position
+  left_pointer, right_pointer = 0, len(arr) - 1
+
+  #loop through the array from both sides toward the middle to find the pair 
+  #initialize a temp_sum variable to keep track of the sum of leftmost and rightmost value
+  #if temp_sum is equal to target_sum, we found the pair and just return the their indices
+  #else if temp_sum is smaller than target_sum, it means the desired pair would possibly be on the right side of left_pointer -> left_pointer += 1
+  #else if temp_sum is larger than target_sum, it means the desired pair would possibly be on the left side of right_pointer -> right_pointer -= 1
+  #if we get out of the while loop without return the actual pair, it means there is no pair sum up to the target_sum, just return None
+
+  while left_pointer < right_pointer:
+    temp_sum =  arr[left_pointer] + arr[right_pointer]
+    if temp_sum == target_sum:
+       return [left_pointer, right_pointer]
+    elif temp_sum > target_sum:
+      right_pointer -= 1 
+    else:
+      left_pointer += 1
+
+  return None
```

Comment:
Love it! The code is easier to read if the psuedo-code is above it, in my opinion.